samcli/commands/init/command.py

```diff
@@ -118,9 +118,13 @@ def wrapped(*args, **kwargs):
             package_type = ctx.params.get("package_type")
             base_image = ctx.params.get("base_image")
             dependency_manager = ctx.params.get("dependency_manager")
+            location = ctx.params.get("location")
 
-            # dependency manager is only required for ZIP package type and for java based IMAGE package types
-            if package_type == ZIP or (base_image and "java" in base_image):
+            java_base_image = base_image and "java" in base_image
```

Comment:
Why are we bleeding this information over? especially checking for a string in a base_image?

Reply:
Sorry not sure if I 100% follow, but the base image option is optional and can has a couple of java options (`amazon/java11-base`, `amazon/java8.al2-base`, `amazon/java8-base`) hence the `in` check.

Reply:
Could you add a fixme/todo with respect to our code organization here, we are looking up an arbitrary string in the base image to determine if its a java image.